lib/mix/lib/mix/dep/loader.ex

```diff
@@ -320,6 +322,13 @@ defmodule Mix.Dep.Loader do
     end
   end
 
+  defp incompatible_requirement?(%Mix.Dep{opts: opts}) do
+    case opts[:lock] do
+      {scm, _, _, _} -> is_nil(opts[scm])
```

Comment:
Thank you @tuvistavie. Don't worry about it as I am working on a fix because it is blocking the 1.2 release. :) This is one of the three issues left.